migrate.go

```diff
@@ -389,34 +395,71 @@ func PlanMigration(db *sql.DB, dialect string, m MigrationSource, dir MigrationD
 	return result, dbMap, nil
 }
 
-// Filter a slice of migrations into ones that should be applied.
-func ToApply(migrations []*Migration, current string, direction MigrationDirection) []*Migration {
-	var index = -1
-	if current != "" {
-		for index < len(migrations)-1 {
-			index++
-			if migrations[index].Id == current {
+// Searches the arrays for the latest elements with common ids.
+// Returns the indexes of the common id.
+func lastCommon(x []*Migration, y []*Migration) (int, int) {
+	xIndexMatch := -1
+	yIndexMatch := 0
+	for i := 0; i < len(x); i++ {
+		existingId := x[i].Id
+		for j := yIndexMatch; j < len(y); j++ {
+			if y[j].Id == existingId {
+				xIndexMatch = i
+				yIndexMatch = j
 				break
 			}
 		}
 	}
+	if xIndexMatch == -1 {
+		return -1, -1 // We never found a match; the arrays have nothing in common
+	}
+	return xIndexMatch, yIndexMatch
+}
 
+// Filter a slice of migrations into ones that should be applied.
+func ToApply(migrations, existingMigrations []*Migration, direction MigrationDirection) []*Migration {
```

Comment:
This is a breaking change to a public method. It doesn't feel like the kind of method that library clients would actually use. However, when PRing upstream, I'll offer to rewrite this as `ToApplySafe` or something similar and avoid breaking the signature of `ToApply()`.